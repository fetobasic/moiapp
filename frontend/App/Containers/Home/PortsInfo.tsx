import React from 'react';
import { useSelector } from 'react-redux';
import { View, ScrollView } from 'react-native';
import { NativeStackScreenProps } from '@react-navigation/native-stack';

import { HeaderSimple as Header, Info } from 'App/Components';

import { ApplicationStyles, Colors } from 'App/Themes';
import { HomeStackParamList } from 'App/Types/NavigationStackParamList';
import renderElement from 'App/Services/RenderElement';
import { getCurrentDevice } from 'App/Store/Devices/selectors';

import InputIcon from 'App/Images/Icons/input.svg';
import OutputIcon from 'App/Images/Icons/output.svg';

import SocketIcon from 'App/Images/Icons/portInSocket.svg';
import SunIcon from 'App/Images/Icons/portInSun.svg';
import AuxIcon from 'App/Images/Icons/portInAux.svg';

import AcIcon from 'App/Images/Icons/portOutAc.svg';
import V12Icon from 'App/Images/Icons/portOut12v.svg';
import UsbIcon from 'App/Images/Icons/portOutUsb.svg';
import { YetiState } from 'App/Types/Yeti';

type Props = NativeStackScreenProps<HomeStackParamList, 'PortsInfo'>;

function PortsInfo({ route }: Props) {
  const device = useSelector(getCurrentDevice(route.params.device.thingName || '')) as YetiState;

  return (
    <View style={ApplicationStyles.mainContainer}>
      <Header title="Ports" />
      <ScrollView>
        {/* Input */}
        <Info
          isFirst
          icon={renderElement(<InputIcon />)}
          title={`${device.wattsIn} W`}
          description="Input"
          titleColor={Colors.blue}
        />
        <Info
          icon={renderElement(<SocketIcon />)}
          title="From the wall"
          description="Fully recharges in 14 hours using the included 120W Power Supply AC Wall Charger."
        />
        <Info
          icon={renderElement(<SunIcon />)}
          title="From the sun"
          description="Recharge from the sun by connecting a compatible solar panel. Charge times are dependent on the size of the solar panel."
        />
        <Info
          icon={renderElement(<AuxIcon />)}
          title="From the car"
          description="The Goal Zero Yeti 1400 can be charged by plugging into your 12V adapter using the Goal Zero Yeti Lithium 12V Car Charging Cable."
        />

        {/* Output */}
        <Info
          isFirst
          icon={renderElement(<OutputIcon />)}
          title={`${device.wattsOut} W`}
          description="Output"
          titleColor={Colors.green}
        />
        <Info
          icon={renderElement(<AcIcon />)}
          title="Take your wall outlet anywhere"
          description="Allows you to run power-hungry devices and appliances with confidence."
        />
        <Info
          icon={renderElement(<V12Icon />)}
          title="Versatile port options"
          description="Power a wide range of devices with seven different port options including fast-charging 60W USB-C Power Delivery, multiple USB-A ports, regulated 12V, and two 120V AC ports."
        />
        <Info icon={renderElement(<UsbIcon />)} title="Connection USB and USB-C" />
      </ScrollView>
    </View>
  );
}

export default PortsInfo;
