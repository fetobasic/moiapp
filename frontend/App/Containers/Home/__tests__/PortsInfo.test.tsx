import React from 'react';
import { render } from '@testing-library/react-native';
import { Provider } from 'react-redux';
import createMockStore from 'redux-mock-store';
import { mockedState, yeti6G, yeti6GLatestFW, yetiLegacy } from 'App/Fixtures/mocks/mockedState';
import PortsInfo from '../PortsInfo';
import { Store } from 'redux';
import { NavigationContainer } from '@react-navigation/native';

jest.useFakeTimers();
const mockStore = createMockStore([]);

const renderWithStore = (store: Store, device = yeti6GLatestFW) => {
  const props = {
    navigation: {
      navigate: jest.fn() as any,
    },
    route: { params: { device } },
  } as any;
  return render(
    <Provider store={store}>
      <NavigationContainer>
        <PortsInfo {...props} />
      </NavigationContainer>
    </Provider>,
  ).toJSON();
};

describe('PortsInfo', () => {
  afterEach(() => {
    jest.clearAllMocks();
  });

  describe('renders screen for Yeti', () => {
    test('renders screen for 6G yeti', () => {
      const updatedState = {
        ...mockedState,
        application: {
          ...mockedState.application,
          lastDevice: yeti6G.thingName,
        },

        devicesInfo: {
          ...mockedState.devicesInfo,
          data: [yeti6G],
        },
      };
      const store = mockStore(updatedState);
      const tree = renderWithStore(store);
      expect(tree).toMatchSnapshot();
    });

    test('renders screen for legacy yeti', () => {
      const updatedState = {
        ...mockedState,
        application: {
          ...mockedState.application,
          lastDevice: yetiLegacy.thingName,
        },
        devicesInfo: {
          ...mockedState.devicesInfo,
          data: [yetiLegacy],
        },
      };
      const store = mockStore(updatedState);
      const tree = renderWithStore(store, yetiLegacy as any);
      expect(tree).toMatchSnapshot();
    });
  });
});
