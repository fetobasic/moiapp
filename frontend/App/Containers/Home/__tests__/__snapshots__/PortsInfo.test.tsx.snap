// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PortsInfo renders screen for Yeti renders screen for 6G yeti 1`] = `
<View
  style={
    {
      "backgroundColor": "rgba(27, 27, 27, 1)",
      "flex": 1,
    }
  }
>
  <View
    style={
      [
        {
          "backgroundColor": "rgba(27, 27, 27, 1)",
          "borderBottomLeftRadius": 24,
          "borderBottomRightRadius": 24,
          "elevation": 12,
          "shadowColor": "rgba(0, 0, 0, 1)",
          "shadowOffset": {
            "height": 4,
            "width": 4,
          },
          "shadowOpacity": 0.2,
          "shadowRadius": 12,
          "zIndex": 1,
        },
        {
          "paddingTop": 0,
        },
      ]
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "height": 60,
          "justifyContent": "center",
          "marginHorizontal": 24,
        }
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        hitSlop={20}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          [
            {
              "opacity": 1,
            },
            {
              "left": 0,
              "padding": 10,
              "position": "absolute",
            },
          ]
        }
      >
        <SvgMock />
      </View>
      <View
        style={
          {
            "alignItems": "center",
            "marginHorizontal": 48,
          }
        }
      >
        <Text
          numberOfLines={1}
          style={
            {
              "color": "rgba(255, 255, 255, 1)",
              "fontFamily": "RobotoCondensed-Medium",
              "fontSize": 20,
              "fontWeight": "500",
              "lineHeight": 32,
            }
          }
        >
          Ports
        </Text>
      </View>
    </View>
  </View>
  <RCTScrollView>
    <View>
      <View
        style={
          [
            {
              "backgroundColor": "rgba(255, 255, 255, 0.09)",
              "paddingHorizontal": 24,
            },
            {
              "marginTop": 16,
            },
          ]
        }
        testID="info"
      >
        <View
          style={
            [
              {
                "borderTopColor": "rgba(255, 255, 255, 0.12)",
                "borderTopWidth": 1,
                "paddingVertical": 16,
              },
              {
                "borderTopWidth": 0,
              },
            ]
          }
        >
          <View
            style={
              {
                "flexDirection": "row",
              }
            }
          >
            <View
              style={
                {
                  "alignItems": "center",
                  "height": 32,
                  "width": 32,
                }
              }
            >
              <SvgMock />
            </View>
            <Text
              style={
                [
                  {
                    "color": "rgba(255, 255, 255, 0.87)",
                    "fontFamily": "Roboto-Medium",
                    "fontSize": 16,
                    "fontWeight": "400",
                    "lineHeight": 24,
                    "marginLeft": 12,
                  },
                  {
                    "color": "rgba(28, 154, 214, 1)",
                    "fontFamily": "RobotoCondensed-Medium",
                    "fontSize": 20,
                    "fontWeight": "500",
                    "lineHeight": 32,
                  },
                ]
              }
            >
              undefined W
            </Text>
          </View>
          <Text
            style={
              {
                "color": "rgba(255, 255, 255, 0.6)",
                "fontFamily": "Roboto-Medium",
                "fontSize": 14,
                "fontWeight": "400",
                "lineHeight": 20,
                "marginLeft": 46,
              }
            }
          >
            Input
          </Text>
        </View>
      </View>
      <View
        style={
          [
            {
              "backgroundColor": "rgba(255, 255, 255, 0.09)",
              "paddingHorizontal": 24,
            },
            undefined,
          ]
        }
        testID="info"
      >
        <View
          style={
            [
              {
                "borderTopColor": "rgba(255, 255, 255, 0.12)",
                "borderTopWidth": 1,
                "paddingVertical": 16,
              },
              undefined,
            ]
          }
        >
          <View
            style={
              {
                "flexDirection": "row",
              }
            }
          >
            <View
              style={
                {
                  "alignItems": "center",
                  "height": 32,
                  "width": 32,
                }
              }
            >
              <SvgMock />
            </View>
            <Text
              style={
                [
                  {
                    "color": "rgba(255, 255, 255, 0.87)",
                    "fontFamily": "Roboto-Medium",
                    "fontSize": 16,
                    "fontWeight": "400",
                    "lineHeight": 24,
                    "marginLeft": 12,
                  },
                  undefined,
                ]
              }
            >
              From the wall
            </Text>
          </View>
          <Text
            style={
              {
                "color": "rgba(255, 255, 255, 0.6)",
                "fontFamily": "Roboto-Medium",
                "fontSize": 14,
                "fontWeight": "400",
                "lineHeight": 20,
                "marginLeft": 46,
              }
            }
          >
            Fully recharges in 14 hours using the included 120W Power Supply AC Wall Charger.
          </Text>
        </View>
      </View>
      <View
        style={
          [
            {
              "backgroundColor": "rgba(255, 255, 255, 0.09)",
              "paddingHorizontal": 24,
            },
            undefined,
          ]
        }
        testID="info"
      >
        <View
          style={
            [
              {
                "borderTopColor": "rgba(255, 255, 255, 0.12)",
                "borderTopWidth": 1,
                "paddingVertical": 16,
              },
              undefined,
            ]
          }
        >
          <View
            style={
              {
                "flexDirection": "row",
              }
            }
          >
            <View
              style={
                {
                  "alignItems": "center",
                  "height": 32,
                  "width": 32,
                }
              }
            >
              <SvgMock />
            </View>
            <Text
              style={
                [
                  {
                    "color": "rgba(255, 255, 255, 0.87)",
                    "fontFamily": "Roboto-Medium",
                    "fontSize": 16,
                    "fontWeight": "400",
                    "lineHeight": 24,
                    "marginLeft": 12,
                  },
                  undefined,
                ]
              }
            >
              From the sun
            </Text>
          </View>
          <Text
            style={
              {
                "color": "rgba(255, 255, 255, 0.6)",
                "fontFamily": "Roboto-Medium",
                "fontSize": 14,
                "fontWeight": "400",
                "lineHeight": 20,
                "marginLeft": 46,
              }
            }
          >
            Recharge from the sun by connecting a compatible solar panel. Charge times are dependent on the size of the solar panel.
          </Text>
        </View>
      </View>
      <View
        style={
          [
            {
              "backgroundColor": "rgba(255, 255, 255, 0.09)",
              "paddingHorizontal": 24,
            },
            undefined,
          ]
        }
        testID="info"
      >
        <View
          style={
            [
              {
                "borderTopColor": "rgba(255, 255, 255, 0.12)",
                "borderTopWidth": 1,
                "paddingVertical": 16,
              },
              undefined,
            ]
          }
        >
          <View
            style={
              {
                "flexDirection": "row",
              }
            }
          >
            <View
              style={
                {
                  "alignItems": "center",
                  "height": 32,
                  "width": 32,
                }
              }
            >
              <SvgMock />
            </View>
            <Text
              style={
                [
                  {
                    "color": "rgba(255, 255, 255, 0.87)",
                    "fontFamily": "Roboto-Medium",
                    "fontSize": 16,
                    "fontWeight": "400",
                    "lineHeight": 24,
                    "marginLeft": 12,
                  },
                  undefined,
                ]
              }
            >
              From the car
            </Text>
          </View>
          <Text
            style={
              {
                "color": "rgba(255, 255, 255, 0.6)",
                "fontFamily": "Roboto-Medium",
                "fontSize": 14,
                "fontWeight": "400",
                "lineHeight": 20,
                "marginLeft": 46,
              }
            }
          >
            The Goal Zero Yeti 1400 can be charged by plugging into your 12V adapter using the Goal Zero Yeti Lithium 12V Car Charging Cable.
          </Text>
        </View>
      </View>
      <View
        style={
          [
            {
              "backgroundColor": "rgba(255, 255, 255, 0.09)",
              "paddingHorizontal": 24,
            },
            {
              "marginTop": 16,
            },
          ]
        }
        testID="info"
      >
        <View
          style={
            [
              {
                "borderTopColor": "rgba(255, 255, 255, 0.12)",
                "borderTopWidth": 1,
                "paddingVertical": 16,
              },
              {
                "borderTopWidth": 0,
              },
            ]
          }
        >
          <View
            style={
              {
                "flexDirection": "row",
              }
            }
          >
            <View
              style={
                {
                  "alignItems": "center",
                  "height": 32,
                  "width": 32,
                }
              }
            >
              <SvgMock />
            </View>
            <Text
              style={
                [
                  {
                    "color": "rgba(255, 255, 255, 0.87)",
                    "fontFamily": "Roboto-Medium",
                    "fontSize": 16,
                    "fontWeight": "400",
                    "lineHeight": 24,
                    "marginLeft": 12,
                  },
                  {
                    "color": "rgba(191, 210, 43, 1)",
                    "fontFamily": "RobotoCondensed-Medium",
                    "fontSize": 20,
                    "fontWeight": "500",
                    "lineHeight": 32,
                  },
                ]
              }
            >
              undefined W
            </Text>
          </View>
          <Text
            style={
              {
                "color": "rgba(255, 255, 255, 0.6)",
                "fontFamily": "Roboto-Medium",
                "fontSize": 14,
                "fontWeight": "400",
                "lineHeight": 20,
                "marginLeft": 46,
              }
            }
          >
            Output
          </Text>
        </View>
      </View>
      <View
        style={
          [
            {
              "backgroundColor": "rgba(255, 255, 255, 0.09)",
              "paddingHorizontal": 24,
            },
            undefined,
          ]
        }
        testID="info"
      >
        <View
          style={
            [
              {
                "borderTopColor": "rgba(255, 255, 255, 0.12)",
                "borderTopWidth": 1,
                "paddingVertical": 16,
              },
              undefined,
            ]
          }
        >
          <View
            style={
              {
                "flexDirection": "row",
              }
            }
          >
            <View
              style={
                {
                  "alignItems": "center",
                  "height": 32,
                  "width": 32,
                }
              }
            >
              <SvgMock />
            </View>
            <Text
              style={
                [
                  {
                    "color": "rgba(255, 255, 255, 0.87)",
                    "fontFamily": "Roboto-Medium",
                    "fontSize": 16,
                    "fontWeight": "400",
                    "lineHeight": 24,
                    "marginLeft": 12,
                  },
                  undefined,
                ]
              }
            >
              Take your wall outlet anywhere
            </Text>
          </View>
          <Text
            style={
              {
                "color": "rgba(255, 255, 255, 0.6)",
                "fontFamily": "Roboto-Medium",
                "fontSize": 14,
                "fontWeight": "400",
                "lineHeight": 20,
                "marginLeft": 46,
              }
            }
          >
            Allows you to run power-hungry devices and appliances with confidence.
          </Text>
        </View>
      </View>
      <View
        style={
          [
            {
              "backgroundColor": "rgba(255, 255, 255, 0.09)",
              "paddingHorizontal": 24,
            },
            undefined,
          ]
        }
        testID="info"
      >
        <View
          style={
            [
              {
                "borderTopColor": "rgba(255, 255, 255, 0.12)",
                "borderTopWidth": 1,
                "paddingVertical": 16,
              },
              undefined,
            ]
          }
        >
          <View
            style={
              {
                "flexDirection": "row",
              }
            }
          >
            <View
              style={
                {
                  "alignItems": "center",
                  "height": 32,
                  "width": 32,
                }
              }
            >
              <SvgMock />
            </View>
            <Text
              style={
                [
                  {
                    "color": "rgba(255, 255, 255, 0.87)",
                    "fontFamily": "Roboto-Medium",
                    "fontSize": 16,
                    "fontWeight": "400",
                    "lineHeight": 24,
                    "marginLeft": 12,
                  },
                  undefined,
                ]
              }
            >
              Versatile port options
            </Text>
          </View>
          <Text
            style={
              {
                "color": "rgba(255, 255, 255, 0.6)",
                "fontFamily": "Roboto-Medium",
                "fontSize": 14,
                "fontWeight": "400",
                "lineHeight": 20,
                "marginLeft": 46,
              }
            }
          >
            Power a wide range of devices with seven different port options including fast-charging 60W USB-C Power Delivery, multiple USB-A ports, regulated 12V, and two 120V AC ports.
          </Text>
        </View>
      </View>
      <View
        style={
          [
            {
              "backgroundColor": "rgba(255, 255, 255, 0.09)",
              "paddingHorizontal": 24,
            },
            undefined,
          ]
        }
        testID="info"
      >
        <View
          style={
            [
              {
                "borderTopColor": "rgba(255, 255, 255, 0.12)",
                "borderTopWidth": 1,
                "paddingVertical": 16,
              },
              undefined,
            ]
          }
        >
          <View
            style={
              {
                "flexDirection": "row",
              }
            }
          >
            <View
              style={
                {
                  "alignItems": "center",
                  "height": 32,
                  "width": 32,
                }
              }
            >
              <SvgMock />
            </View>
            <Text
              style={
                [
                  {
                    "color": "rgba(255, 255, 255, 0.87)",
                    "fontFamily": "Roboto-Medium",
                    "fontSize": 16,
                    "fontWeight": "400",
                    "lineHeight": 24,
                    "marginLeft": 12,
                  },
                  undefined,
                ]
              }
            >
              Connection USB and USB-C
            </Text>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;

exports[`PortsInfo renders screen for Yeti renders screen for legacy yeti 1`] = `
<View
  style={
    {
      "backgroundColor": "rgba(27, 27, 27, 1)",
      "flex": 1,
    }
  }
>
  <View
    style={
      [
        {
          "backgroundColor": "rgba(27, 27, 27, 1)",
          "borderBottomLeftRadius": 24,
          "borderBottomRightRadius": 24,
          "elevation": 12,
          "shadowColor": "rgba(0, 0, 0, 1)",
          "shadowOffset": {
            "height": 4,
            "width": 4,
          },
          "shadowOpacity": 0.2,
          "shadowRadius": 12,
          "zIndex": 1,
        },
        {
          "paddingTop": 0,
        },
      ]
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "height": 60,
          "justifyContent": "center",
          "marginHorizontal": 24,
        }
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        hitSlop={20}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          [
            {
              "opacity": 1,
            },
            {
              "left": 0,
              "padding": 10,
              "position": "absolute",
            },
          ]
        }
      >
        <SvgMock />
      </View>
      <View
        style={
          {
            "alignItems": "center",
            "marginHorizontal": 48,
          }
        }
      >
        <Text
          numberOfLines={1}
          style={
            {
              "color": "rgba(255, 255, 255, 1)",
              "fontFamily": "RobotoCondensed-Medium",
              "fontSize": 20,
              "fontWeight": "500",
              "lineHeight": 32,
            }
          }
        >
          Ports
        </Text>
      </View>
    </View>
  </View>
  <RCTScrollView>
    <View>
      <View
        style={
          [
            {
              "backgroundColor": "rgba(255, 255, 255, 0.09)",
              "paddingHorizontal": 24,
            },
            {
              "marginTop": 16,
            },
          ]
        }
        testID="info"
      >
        <View
          style={
            [
              {
                "borderTopColor": "rgba(255, 255, 255, 0.12)",
                "borderTopWidth": 1,
                "paddingVertical": 16,
              },
              {
                "borderTopWidth": 0,
              },
            ]
          }
        >
          <View
            style={
              {
                "flexDirection": "row",
              }
            }
          >
            <View
              style={
                {
                  "alignItems": "center",
                  "height": 32,
                  "width": 32,
                }
              }
            >
              <SvgMock />
            </View>
            <Text
              style={
                [
                  {
                    "color": "rgba(255, 255, 255, 0.87)",
                    "fontFamily": "Roboto-Medium",
                    "fontSize": 16,
                    "fontWeight": "400",
                    "lineHeight": 24,
                    "marginLeft": 12,
                  },
                  {
                    "color": "rgba(28, 154, 214, 1)",
                    "fontFamily": "RobotoCondensed-Medium",
                    "fontSize": 20,
                    "fontWeight": "500",
                    "lineHeight": 32,
                  },
                ]
              }
            >
              0 W
            </Text>
          </View>
          <Text
            style={
              {
                "color": "rgba(255, 255, 255, 0.6)",
                "fontFamily": "Roboto-Medium",
                "fontSize": 14,
                "fontWeight": "400",
                "lineHeight": 20,
                "marginLeft": 46,
              }
            }
          >
            Input
          </Text>
        </View>
      </View>
      <View
        style={
          [
            {
              "backgroundColor": "rgba(255, 255, 255, 0.09)",
              "paddingHorizontal": 24,
            },
            undefined,
          ]
        }
        testID="info"
      >
        <View
          style={
            [
              {
                "borderTopColor": "rgba(255, 255, 255, 0.12)",
                "borderTopWidth": 1,
                "paddingVertical": 16,
              },
              undefined,
            ]
          }
        >
          <View
            style={
              {
                "flexDirection": "row",
              }
            }
          >
            <View
              style={
                {
                  "alignItems": "center",
                  "height": 32,
                  "width": 32,
                }
              }
            >
              <SvgMock />
            </View>
            <Text
              style={
                [
                  {
                    "color": "rgba(255, 255, 255, 0.87)",
                    "fontFamily": "Roboto-Medium",
                    "fontSize": 16,
                    "fontWeight": "400",
                    "lineHeight": 24,
                    "marginLeft": 12,
                  },
                  undefined,
                ]
              }
            >
              From the wall
            </Text>
          </View>
          <Text
            style={
              {
                "color": "rgba(255, 255, 255, 0.6)",
                "fontFamily": "Roboto-Medium",
                "fontSize": 14,
                "fontWeight": "400",
                "lineHeight": 20,
                "marginLeft": 46,
              }
            }
          >
            Fully recharges in 14 hours using the included 120W Power Supply AC Wall Charger.
          </Text>
        </View>
      </View>
      <View
        style={
          [
            {
              "backgroundColor": "rgba(255, 255, 255, 0.09)",
              "paddingHorizontal": 24,
            },
            undefined,
          ]
        }
        testID="info"
      >
        <View
          style={
            [
              {
                "borderTopColor": "rgba(255, 255, 255, 0.12)",
                "borderTopWidth": 1,
                "paddingVertical": 16,
              },
              undefined,
            ]
          }
        >
          <View
            style={
              {
                "flexDirection": "row",
              }
            }
          >
            <View
              style={
                {
                  "alignItems": "center",
                  "height": 32,
                  "width": 32,
                }
              }
            >
              <SvgMock />
            </View>
            <Text
              style={
                [
                  {
                    "color": "rgba(255, 255, 255, 0.87)",
                    "fontFamily": "Roboto-Medium",
                    "fontSize": 16,
                    "fontWeight": "400",
                    "lineHeight": 24,
                    "marginLeft": 12,
                  },
                  undefined,
                ]
              }
            >
              From the sun
            </Text>
          </View>
          <Text
            style={
              {
                "color": "rgba(255, 255, 255, 0.6)",
                "fontFamily": "Roboto-Medium",
                "fontSize": 14,
                "fontWeight": "400",
                "lineHeight": 20,
                "marginLeft": 46,
              }
            }
          >
            Recharge from the sun by connecting a compatible solar panel. Charge times are dependent on the size of the solar panel.
          </Text>
        </View>
      </View>
      <View
        style={
          [
            {
              "backgroundColor": "rgba(255, 255, 255, 0.09)",
              "paddingHorizontal": 24,
            },
            undefined,
          ]
        }
        testID="info"
      >
        <View
          style={
            [
              {
                "borderTopColor": "rgba(255, 255, 255, 0.12)",
                "borderTopWidth": 1,
                "paddingVertical": 16,
              },
              undefined,
            ]
          }
        >
          <View
            style={
              {
                "flexDirection": "row",
              }
            }
          >
            <View
              style={
                {
                  "alignItems": "center",
                  "height": 32,
                  "width": 32,
                }
              }
            >
              <SvgMock />
            </View>
            <Text
              style={
                [
                  {
                    "color": "rgba(255, 255, 255, 0.87)",
                    "fontFamily": "Roboto-Medium",
                    "fontSize": 16,
                    "fontWeight": "400",
                    "lineHeight": 24,
                    "marginLeft": 12,
                  },
                  undefined,
                ]
              }
            >
              From the car
            </Text>
          </View>
          <Text
            style={
              {
                "color": "rgba(255, 255, 255, 0.6)",
                "fontFamily": "Roboto-Medium",
                "fontSize": 14,
                "fontWeight": "400",
                "lineHeight": 20,
                "marginLeft": 46,
              }
            }
          >
            The Goal Zero Yeti 1400 can be charged by plugging into your 12V adapter using the Goal Zero Yeti Lithium 12V Car Charging Cable.
          </Text>
        </View>
      </View>
      <View
        style={
          [
            {
              "backgroundColor": "rgba(255, 255, 255, 0.09)",
              "paddingHorizontal": 24,
            },
            {
              "marginTop": 16,
            },
          ]
        }
        testID="info"
      >
        <View
          style={
            [
              {
                "borderTopColor": "rgba(255, 255, 255, 0.12)",
                "borderTopWidth": 1,
                "paddingVertical": 16,
              },
              {
                "borderTopWidth": 0,
              },
            ]
          }
        >
          <View
            style={
              {
                "flexDirection": "row",
              }
            }
          >
            <View
              style={
                {
                  "alignItems": "center",
                  "height": 32,
                  "width": 32,
                }
              }
            >
              <SvgMock />
            </View>
            <Text
              style={
                [
                  {
                    "color": "rgba(255, 255, 255, 0.87)",
                    "fontFamily": "Roboto-Medium",
                    "fontSize": 16,
                    "fontWeight": "400",
                    "lineHeight": 24,
                    "marginLeft": 12,
                  },
                  {
                    "color": "rgba(191, 210, 43, 1)",
                    "fontFamily": "RobotoCondensed-Medium",
                    "fontSize": 20,
                    "fontWeight": "500",
                    "lineHeight": 32,
                  },
                ]
              }
            >
              0 W
            </Text>
          </View>
          <Text
            style={
              {
                "color": "rgba(255, 255, 255, 0.6)",
                "fontFamily": "Roboto-Medium",
                "fontSize": 14,
                "fontWeight": "400",
                "lineHeight": 20,
                "marginLeft": 46,
              }
            }
          >
            Output
          </Text>
        </View>
      </View>
      <View
        style={
          [
            {
              "backgroundColor": "rgba(255, 255, 255, 0.09)",
              "paddingHorizontal": 24,
            },
            undefined,
          ]
        }
        testID="info"
      >
        <View
          style={
            [
              {
                "borderTopColor": "rgba(255, 255, 255, 0.12)",
                "borderTopWidth": 1,
                "paddingVertical": 16,
              },
              undefined,
            ]
          }
        >
          <View
            style={
              {
                "flexDirection": "row",
              }
            }
          >
            <View
              style={
                {
                  "alignItems": "center",
                  "height": 32,
                  "width": 32,
                }
              }
            >
              <SvgMock />
            </View>
            <Text
              style={
                [
                  {
                    "color": "rgba(255, 255, 255, 0.87)",
                    "fontFamily": "Roboto-Medium",
                    "fontSize": 16,
                    "fontWeight": "400",
                    "lineHeight": 24,
                    "marginLeft": 12,
                  },
                  undefined,
                ]
              }
            >
              Take your wall outlet anywhere
            </Text>
          </View>
          <Text
            style={
              {
                "color": "rgba(255, 255, 255, 0.6)",
                "fontFamily": "Roboto-Medium",
                "fontSize": 14,
                "fontWeight": "400",
                "lineHeight": 20,
                "marginLeft": 46,
              }
            }
          >
            Allows you to run power-hungry devices and appliances with confidence.
          </Text>
        </View>
      </View>
      <View
        style={
          [
            {
              "backgroundColor": "rgba(255, 255, 255, 0.09)",
              "paddingHorizontal": 24,
            },
            undefined,
          ]
        }
        testID="info"
      >
        <View
          style={
            [
              {
                "borderTopColor": "rgba(255, 255, 255, 0.12)",
                "borderTopWidth": 1,
                "paddingVertical": 16,
              },
              undefined,
            ]
          }
        >
          <View
            style={
              {
                "flexDirection": "row",
              }
            }
          >
            <View
              style={
                {
                  "alignItems": "center",
                  "height": 32,
                  "width": 32,
                }
              }
            >
              <SvgMock />
            </View>
            <Text
              style={
                [
                  {
                    "color": "rgba(255, 255, 255, 0.87)",
                    "fontFamily": "Roboto-Medium",
                    "fontSize": 16,
                    "fontWeight": "400",
                    "lineHeight": 24,
                    "marginLeft": 12,
                  },
                  undefined,
                ]
              }
            >
              Versatile port options
            </Text>
          </View>
          <Text
            style={
              {
                "color": "rgba(255, 255, 255, 0.6)",
                "fontFamily": "Roboto-Medium",
                "fontSize": 14,
                "fontWeight": "400",
                "lineHeight": 20,
                "marginLeft": 46,
              }
            }
          >
            Power a wide range of devices with seven different port options including fast-charging 60W USB-C Power Delivery, multiple USB-A ports, regulated 12V, and two 120V AC ports.
          </Text>
        </View>
      </View>
      <View
        style={
          [
            {
              "backgroundColor": "rgba(255, 255, 255, 0.09)",
              "paddingHorizontal": 24,
            },
            undefined,
          ]
        }
        testID="info"
      >
        <View
          style={
            [
              {
                "borderTopColor": "rgba(255, 255, 255, 0.12)",
                "borderTopWidth": 1,
                "paddingVertical": 16,
              },
              undefined,
            ]
          }
        >
          <View
            style={
              {
                "flexDirection": "row",
              }
            }
          >
            <View
              style={
                {
                  "alignItems": "center",
                  "height": 32,
                  "width": 32,
                }
              }
            >
              <SvgMock />
            </View>
            <Text
              style={
                [
                  {
                    "color": "rgba(255, 255, 255, 0.87)",
                    "fontFamily": "Roboto-Medium",
                    "fontSize": 16,
                    "fontWeight": "400",
                    "lineHeight": 24,
                    "marginLeft": 12,
                  },
                  undefined,
                ]
              }
            >
              Connection USB and USB-C
            </Text>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
